pkg/apis/kubermatic/v1/preset.go

```diff
@@ -140,17 +156,38 @@ func (s Hetzner) IsValid() bool {
 type Azure struct {
 	ProviderPreset `json:",inline"`
 
-	TenantID       string `json:"tenantID"`
+	// The Azure Active Directory Tenant used for the user cluster.
+	TenantID string `json:"tenantID"`
+	// The Azure Subscription used for the user cluster.
 	SubscriptionID string `json:"subscriptionID"`
-	ClientID       string `json:"clientID"`
-	ClientSecret   string `json:"clientSecret"`
-
-	ResourceGroup     string `json:"resourceGroup,omitempty"`
+	// The service principal used to access Azure.
+	ClientID string `json:"clientID"`
+	// The client secret corresponding to the given service principal.
+	ClientSecret string `json:"clientSecret"`
+
+	// The resource group that will be used to look up and create resources for the cluster in.
+	// If set to empty string at cluster creation, a new resource group will be created and this field will be updated to
+	// the generated resource group's name.
+	ResourceGroup string `json:"resourceGroup,omitempty"`
+	// Optional: Defines a second resource group that will be used for VNet related resources instead.
+	// If left empty, NO additional resource group will be created and all VNet related resources use the resource group defined by `resourceGroup`.
 	VNetResourceGroup string `json:"vnetResourceGroup,omitempty"`
-	VNetName          string `json:"vnet,omitempty"`
-	SubnetName        string `json:"subnet,omitempty"`
-	RouteTableName    string `json:"routeTable,omitempty"`
-	SecurityGroup     string `json:"securityGroup,omitempty"`
+	// The name of the VNet resource used for setting up networking in.
+	// If set to empty string at cluster creation, a new VNet will be created and this field will be updated to
+	// the generated VNet's name.
+	VNetName string `json:"vnet,omitempty"`
+	// The name of a subnet in the VNet referenced by `vnet`.
+	// If set to empty string at cluster creation, a new subnet will be created and this field will be updated to
+	// the generated subnet's name. If no VNet is defined at cluster creation, this field should be empty as well.
+	SubnetName string `json:"subnet,omitempty"`
+	// The name of a route table associated with the subnet referenced by `subnet`.
+	// If set to empty string at cluster creation, a new route table will be created and this field will be updated to
+	// the generated route table's name. If no subnet is defined at cluster creation, this field should be empty as well.
+	RouteTableName string `json:"routeTable,omitempty"`
+	// The name of a security group associated with the subnet referenced by `subnet`.
+	// If set to empty string at cluster creation, a new security group will be created and this field will be updated to
+	// the generated security group's name. If no subnet is defined at cluster creation, this field should be empty as well.
+	SecurityGroup string `json:"securityGroup,omitempty"`
```

Comment:
These are lovely descriptions. So many of ours are just one sentence that is usually not much more than the field name itself, but these here offer real benefit to the end user. Nice.